nextcord/ui/item.py

```diff
@@ -117,3 +128,26 @@ async def callback(self, interaction: Interaction) -> None:
         interaction: :class:`.Interaction`
             The interaction that triggered this UI item.
         """
+
+
+class ModalItem(Item, Generic[ModalT_co]):
+    """Represents the Modal UI item that all UI components supporting :class:`Modal` inherit from.
+
+    The current UI items supported for Modal are:
+
+    - :class:`nextcord.ui.TextInput`
+
+    .. versionadded:: 2.6
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self._modal: Optional[ModalT_co] = None
+
+        # backwards compatibility
+        self.callback: Literal[None] = None
```

Comment:
If this is slated for `3.x` which I think it should, backwards compat is not as much of an issue